File: AndroidStudioProjects/AlertDialog/app/src/main/java/com/ecrinaybay/alertdialog/MainActivity.java
package com.ecrinaybay.alertdialog;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Context
        //Activity Context -->Aktiviteyle alakalı durumlarda this kullanıyoruz.
        //this
        //App Context --> Genel app'i ilgilendiren durumlarda getApplicationContext()'i kullanıyoruz. Genelde app ile alakalı olanlar activitiy'i de ilgilendirebilir.
        //getApplicationContext()

        Toast.makeText(this,"Toast Message",Toast.LENGTH_LONG).show(); //Burada MainActivity.this yerine sadece this kullanabilriz
        //Toast.makeText(getApplicationContext(),"Toast Message",Toast.LENGTH_LONG).show(); //Bu şekilde de kullanabiliriz.
    }

    public void save(View view){
        AlertDialog.Builder alert = new AlertDialog.Builder(this);
        alert.setTitle("Save");
        alert.setMessage("Are you sure?");
        alert.setPositiveButton("Yes", new DialogInterface.OnClickListener() { //listeneri bir olay olduğununda ne yapılacağını yazdığımız dinleyici metotlar , arayüzler olarak düşünebiliriz.
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                //save
                Toast.makeText(MainActivity.this,"Saved",Toast.LENGTH_LONG).show(); //Burada MainActivity.this yerine sadece this kullanamayız.
                //Toast.makeText(getApplicationContext(),"Saved",Toast.LENGTH_LONG).show(); //Bu şekilde de yapabiliriz.
            }
        });
        alert.setNegativeButton("No", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                Toast.makeText(MainActivity.this,"Not Saved",Toast.LENGTH_LONG).show();
            }
        });
        alert.show();

    }


}

File: AndroidStudioProjects/AlertDialogKotlin/app/src/main/java/com/ecrinaybay/alertdialogkotlin/MainActivity.kt
package com.ecrinaybay.alertdialogkotlin

import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 1)Toast 2) AlertDialog 3) Snackbar

        // Context
        //Activity Context -> this
        //Application Context -> applicationContext

        //Toast.makeText(applicationContext, "Welcome!", Toast.LENGTH_LONG).show()
        //Toast.makeText(this, "Welcome!", Toast.LENGTH_LONG).show()
        Toast.makeText(this@MainActivity, "Welcome!", Toast.LENGTH_LONG).show()



    }

    fun save( view : View){

        val alert = AlertDialog.Builder(this@MainActivity)
        alert.setTitle("Save")
        alert.setMessage("Are you sure?")

        alert.setPositiveButton("Yes", object : DialogInterface.OnClickListener{
            override fun onClick(dialog: DialogInterface?, which: Int) {
                Toast.makeText(this@MainActivity,"Saved!",Toast.LENGTH_LONG).show()
            }
        })

        alert.setNegativeButton("No"){ dialog, which ->
            Toast.makeText(this@MainActivity,"Not Saved!",Toast.LENGTH_LONG).show()
        }

        alert.show()

    }
}




























